docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,202 @@
+# Semantics of destructive operations
+
+## Summary
+
+This design document stems from the initial discussion on the [original Github issue #82](https://github.com/aws-controllers-k8s/community/issues/82).
+
+ACK currently treats all resources with the same amount of protection for deletion. That is, if you delete an ACK custom resource, the underlying AWS resource is deleted as well. While this might be the expected behaviour for the vast majority of resources, it also increases the chance that an AWS resource is accidentally deleted - through prematurely or accidentally deleting the ACK CR. Some sort of deletion protection should be introduced to limit the odds of this happening, requiring the user to unset a property manually before the controller will continue with deletion.
+
+Another consideration of destructive operations is that some resources should continue to live on after the ACK CR has been deleted. Similar to the CDK[`RemovalPolicy` type](https://docs.aws.amazon.com/cdk/api/v2//docs/aws-cdk-lib.RemovalPolicy.html#members) and the native [K8s PV reclaim policy](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reclaiming), ACK should support the ability to leave the AWS resource alive while still deleting the custom resource object. Users typically use these removal policy options when working with stateful data, such as S3, EBS and RDS resources, in cases where they would like to migrate the management of the resources from one system to another without having to recreate the data.
```

Comment:
> ACK should support the ability to leave the AWS resource alive while still deleting the custom resource object.

Sure, agree on this one, and this should basically be the reverse of AdoptedResource, IMHO.